src/firmware/main.cpp

```diff
@@ -0,0 +1,69 @@
+#include "SPIFFS.h"
+#include <WiFi.h>
+#include <WebServer.h>
```

Comment:
where does this include and the one above come from?

Reply:
they are from ESP32 sdk, you get them when you are enable device support in your IDE